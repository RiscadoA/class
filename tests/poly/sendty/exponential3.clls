proc main() {
    cut {
        println("!"); !x(y);
        println("sendty"); sendty y(lint);
        println("send 1"); send y(a. println("let"); let a 3);
        println("send 2"); send y(b. println(b); ());
        println("close"); close y
        |x: ? recvty T; recv ~T; recv T; wait |
        println("?"); ?x;
        println("call"); call x(y);
        println("recvty"); recvty y(T);
        println("recv 1"); recv y(a);
        println("recv 2"); recv y(b);
        println("wait"); wait y;
        println("fwd"); fwd a b
    }
};;
