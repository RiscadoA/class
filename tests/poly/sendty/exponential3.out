!
sendty
send 1
let
send 2
close
?
call
recvty
recv 1
recv 2
wait
fwd
3
